docs/src/rule_author/superpowers/mutation_support.md

```diff
@@ -0,0 +1,82 @@
+    Expect breaking changes in minor versions while this remains.
+    It is not suitable for general use unless you are prepared to modify how you are using it each minor release.
+    It is thus suggested that if you are using it to use _tilde_ bounds on supported minor versions.
+
+
+## `MutableTangent`
+The [`MutableTangent`](@ref) type is designed to be a partner to the [`Tangent`](@ref) type, with specific support for being mutated in place.
+It is required to be a structural tangent, having one tangent for each field of the primal object.
+
+Technically, not all `mutable struct`s need to use `MutableTangent` to represent their tangents.
+Just like not all `struct`s need to use `Tangent`s.
+Common examples away from this are natural tangent types like for arrays.
+However, if one is setting up to use a custom tangent type for this it is sufficiently off the beaten path that we can not provide much guidance.
+
+## `zero_tangent`
+
+The [`zero_tangent`](@ref) function functions to give you a zero (i.e. additive identity) for any primal value.
+The [`ZeroTangent`](@ref) type also does this.
+The difference is that [`zero_tangent`](@ref) is in general full structural tangent mirroring the structure of the primal.
+To be technical the promise of [`zero_tangent`](@ref) is that it will be a value that supports mutation.
+However, in practice[^1] this is achieved through in a structural tangent
+For mutation support this is important, since it means that there is mutable memory available in the tangent to be mutated when the primal changes.
+To support this you thus need to make sure your zeros are created in various places with [`zero_tangent`](@ref) rather than []`ZeroTangent`](@ref).
+
+
+
+It is also useful for reasons of type stability, since it forces a consistent type (generally a structural tangent) for any given primal type.
+For this reason AD system implementors might chose to use this to create the tangent for all literal values they encounter, mutable or not,
+and to process the output of `frule`s to convert [`ZeroTangent`](@ref) into corresponding [`zero_tangent`](@ref)s.
+
+## Writing a frule for a mutating function
+It is relatively straight forward to write a frule for a mutating function.
+There are a few key points to follow:
+ - There must be a mutable tangent input for every mutated primal input
+ - When the primal value is changed, the corresponding change must be made to its tangent partner
+ - When a value is returned, return its partnered tangent.
+
+
+### Example
+For example, consider the primal function with:
+1. takes two `Ref`s
+2. doubles the first one in place
+3. overwrites the second one's value with the literal 5.0
+4. returns the first one
+
+
+```julia
+function foo!(a::Base.RefValue, b::Base.RefValue)
+    a[] *= 2
+    b[] = 5.0
+    return a
+end
+```
+
+The frule for this would be:
+```julia
+function ChainRulesCore.frule((ȧ, ḃ), ::typeof(foo!), a::Base.RefValue, b::Base.RefValue)
```

Comment:
No element of the tangent tuple for `foo!`? Maybe I'm just misremembering what the state of ChainRules is for forwards-mode.

Reply:
good catch